audio/out/ao_aaudio.c

```diff
@@ -0,0 +1,423 @@
+    aaudio_result_t (*AAudioStream_requestFlush)(AAudioStream *stream);
+    aaudio_result_t (*AAudioStream_requestStop)(AAudioStream *stream);
+
+    /* Stream query functions */
+    // API 26
+    aaudio_stream_state_t (*AAudioStream_getState)(AAudioStream *stream);
+    aaudio_result_t (*AAudioStream_waitForStateChange)(AAudioStream *stream,
+                                                       aaudio_stream_state_t inputState,
+                                                       aaudio_stream_state_t *nextState,
+                                                       int64_t timeoutNanoseconds);
+    aaudio_result_t (*AAudioStream_read)(AAudioStream *stream,
+                                         void *buffer,
+                                         int32_t numFrames,
+                                         int64_t timeoutNanoseconds);
+    aaudio_result_t (*AAudioStream_write)(AAudioStream *stream, const void *buffer,
+                                          int32_t numFrames,
+                                          int64_t timeoutNanoseconds);
+    aaudio_result_t (*AAudioStream_setBufferSizeInFrames)(AAudioStream *stream,
+                                                          int32_t numFrames);
+    int32_t (*AAudioStream_getBufferSizeInFrames)(AAudioStream *stream);
+    int32_t (*AAudioStream_getFramesPerBurst)(AAudioStream *stream);
+    int32_t (*AAudioStream_getBufferCapacityInFrames)(AAudioStream *stream);
+    int32_t (*AAudioStream_getFramesPerDataCallback)(AAudioStream *stream);
+    int32_t (*AAudioStream_getXRunCount)(AAudioStream *stream);
+    int32_t (*AAudioStream_getSampleRate)(AAudioStream *stream);
+    int32_t (*AAudioStream_getChannelCount)(AAudioStream *stream);
+    int32_t (*AAudioStream_getSamplesPerFrame)(AAudioStream *stream);
+    int32_t (*AAudioStream_getDeviceId)(AAudioStream *stream);
+    aaudio_format_t (*AAudioStream_getFormat)(AAudioStream *stream);
+    aaudio_sharing_mode_t (*AAudioStream_getSharingMode)(AAudioStream *stream);
+    aaudio_performance_mode_t (*AAudioStream_getPerformanceMode)(AAudioStream *stream);
+    aaudio_direction_t (*AAudioStream_getDirection)(AAudioStream *stream);
+    int64_t (*AAudioStream_getFramesWritten)(AAudioStream *stream);
+    int64_t (*AAudioStream_getFramesRead)(AAudioStream *stream);
+    int64_t (*AAudioStream_getTimestamp)(AAudioStream *stream, clockid_t clockid,
+                                         int64_t* framePosition,
+                                         int64_t* timeNanoseconds);
+
+    /* Utility functions */
+    // API 26
+    const char * (*AAudio_convertResultToText)(aaudio_result_t returnCode);
+    const char* (*AAudio_convertStreamStateToText)(aaudio_stream_state_t state);
+};
+struct function_map {
+    const char *symbol;
+    int offset;
+};
+static const struct function_map lib_functions26[] = {
+    {"AAudio_convertResultToText", offsetof(struct priv, AAudio_convertResultToText)},
```

Comment:
Could you make it similar to https://github.com/mpv-player/mpv/blob/d37405c8f33e845a6dc5c60ac38a13a85ae33439/video/vdpau.c#L72-L77 and https://github.com/mpv-player/mpv/blob/d37405c8f33e845a6dc5c60ac38a13a85ae33439/video/vdpau.h#L39-L43

So you declare list of functions once.